File: Python/Basic/collatz.py
def collatz(n):
    len=0
    if n==1:
        print("The length of the sequence is",1)
    while n>1:
        while n%2==1:
            n=int(n*3+1)
            len+=1
            print(n,end=" ")
            if n==1:
                break
        while n%2==0:
            n=int(n/2)
            len+=1
            print(n,end=" ")
            if n==1:
                break
    return len
n=int(input("Enter the value of n"))
ans=collatz(n)
print("\n",ans)

File: Java/src/com/vaishakh/Basic/collatz.java
package com.vaishakh.Basic;

import java.util.Scanner;

public class collatz {
    public static int collatz(int n){
        int len=0;
        if(n==1){
            System.out.println("The length of the sequence is 1");
        }
        while (n>1) {
            while (n % 2 == 1) {
                n = n * 3 + 1;
                len += 1;
                System.out.print(" " + n);
                if (n == 1) {
                    break;
                }
            }
            while (n % 2 == 0) {
                n = n / 2;
                System.out.print(" "+n);
                len += 1;
                if (n == 1) {
                    break;
                }
            }
        }
        return len;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the value of n");
        int n = sc.nextInt();
        int ans=collatz(n);
        System.out.println();
        System.out.println(ans);
    }
}
